lua/telescope/actions/init.lua

```diff
@@ -805,6 +805,213 @@ actions.cycle_previewers_prev = function(prompt_bufnr)
+  opts.column_indent = table.concat(utils.repeated_table(utils.get_default(opts.column_indent, 4), " "))
+  opts.line_padding = utils.get_default(opts.line_padding, 1)
+  opts.separator = utils.get_default(opts.separator, " -> ")
+  opts.close_with_action = utils.get_default(opts.close_with_action, true)
+
+  local find_keymap_buf = function()
+    local bufs = vim.api.nvim_list_bufs()
+    for _, buf in ipairs(bufs) do
+      if string.find(vim.api.nvim_buf_get_name(buf), "_TelescopePickersKeymap") then
+        return buf
+      end
+    end
+  end
+
+  -- close on repeated keypress
+  local km_buf = find_keymap_buf()
+  if km_buf ~= nil then
+    vim.api.nvim_buf_delete(km_buf, { force = true, unload = false })
+    local win_ids = vim.fn.win_findbuf(km_buf)
+    for _, win_id in ipairs(win_ids) do
+      pcall(vim.api.nvim_win_close, win_id, true)
+    end
+    return
+  end
+
+  local entry_display = require "telescope.pickers.entry_display"
+  local displayer = entry_display.create {
+    separator = utils.get_default(opts.seperator, " -> "),
+    items = {
+      { width = opts.mode_with },
+      { width = opts.keybind_width },
+      { width = opts.name_width },
+    },
+  }
+
+  local make_display = function(mapping)
+    return displayer {
+      { mapping.mode, utils.get_default(opts.mode_hl, "TelescopeResultsConstant") },
+      { mapping.keybind, utils.get_default(opts.keybind_hl, "TelescopeResultsVariable") },
+      { mapping.name, utils.get_default(opts.name_hl, "TelescopeResultsFunction") },
+    }
+  end
+
+  local prompt_buf_mappings = __TelescopeKeymapStore[prompt_bufnr]
+
+  local mappings = {}
+  for _, v in pairs(prompt_buf_mappings) do
+    if type(v) == "table" then
+      local name = ""
+      for _, action in ipairs(v) do
+        if type(action) == "string" then
+          name = name == "" and action or name .. " + " .. action
+        end
+      end
+      table.insert(mappings, { mode = v.mode, keybind = v.keybind, name = name })
+    end
+  end
+  if opts.only_show_current_mode then
+    local mode = vim.api.nvim_get_mode().mode
+    mappings = vim.tbl_filter(function(t)
+      return t.mode == mode
+    end, mappings)
+  end
+
+  table.sort(mappings, function(x, y)
+    if x.name < y.name then
+      return true
+    elseif x.name == y.name then
+      -- show normal mode as the standard mode first
+      if x.mode > y.mode then
+        return true
+      else
+        -- loosely prefers simpler mappings
+        if x.keybind < y.keybind then
+          return true
+        end
+        return false
+      end
+    else
+      return false
+    end
+  end)
+
+  local entry_width = #opts.column_padding
+    + opts.mode_width
+    + opts.keybind_width
+    + opts.name_width
+    + (3 * #opts.separator)
+  local num_total_columns = math.floor((vim.o.columns - #opts.column_indent) / entry_width)
+  opts.num_rows = math.ceil(math.min(#mappings / num_total_columns, opts.max_height * vim.o.lines))
```

Comment:
I think it would be better to use `resolver.resolve_height` here, as it offers more configurability for users, and does roughly the same thing.

Reply:
Not so familiar with that code, would that be something like the below?

```lua
  opts.num_rows = math.ceil(
    math.min(#mappings / num_total_columns, resolver.resolve_height(opts.max_height)(_, _, vim.o.lines))
  )
```

Reply:
This looks good to me 👍